### ISSUE TITLE
Quest Extension: Per-Quest Permission nodes

### ISSUE THREAD

**Issue by Ollieboy** [CitizensDev/Citizens#603]
I would like to see citizens.quester.use.quests.(quest name) in the quest permissions, so quests could be chained seamlessly. Ideally it would also not display quests that a user does not have permission for if possible, and upon finishing a quest before it, the reward: permission node could grant that quest permission to the player.

They could also give a negated permission node, -citizens...(quest name) upon completion of that quest, so quests could only be completed once per-player, while still being infinitely repeatable. This would allow for awesome series.

**Comment by Farwaykorse** [comment]
If I understand correctly (small chance) you want to:
1. chain different quests .. like when a player finishes a quest he gets the next quest?
2. be able to have a certain quest only available if a player has finished a particular quest?
3. not to display quest that a user can't play yet.

Well, for 1 and 2 there is a way
3. I'm not sure if if questers show you the quest if you don't meet the requirements.

---

.1. give the next quest as a reward:

```
rewards:
    '0':
        type: quest
        quest: [text questname]
        take: [boolean]         # optional, rewards default false
                                # if false: schedules a new quest
                                # if true: resets Progress (if this is the players current quest)
```

---

.2. add a requirement for a quest:

```
requirements:
    '0':
        type: quest
        quest: [text questname]
        times: [integer]             # optional, default 1, times the required quest needs to be completed.
```

**Comment by Ollieboy** [comment]
Thank you. I did a similar thing with Permission nodes. My main concern was with people being able to repeat quests. I want a system where anybody can do a quest once, and not have it disappear, so I set it to repeats: -1 and then make a permission node required, to which I negate at the end as a reward so they cannot repeat the quest again, but another player can do it. 3. Is still a concern, but minor.

Thanks again, if I played with it a little bit more I wouldn't have need to bug you guys :P. Have a good one.
